text/0095-cognito-construct-library/working-backwards-readme.md

```diff
@@ -0,0 +1,567 @@
+for more details.
+
+Use the `takesPrecendenceOver()` API to set the precedence of one group over the other.
+
+> These are the defaults that will be part of tsdoc, and not part of the README -
+> * description - empty
+> * precedence - `undefined`
+
+## Identity Pools (Federated Identity)
+
+Identity pools enables creation of unique identities for their users and federate them with identity providers. These
+identity providers can be a Cognito user pool, Facebook, Google, or any other SAML-based identity provider.
+With an identity pool, you can obtain temporary, limited-privilege AWS credentials to access other AWS services.
+
+Using the CDK, a new identity pool can be created as part of the stack using the construct's constructor. You must
+specify the `identityPoolName`. This is the quickest way to create an identity pool using all of the defaults -
+
+```ts
+new IdentityPool(this, 'myidentitypool', {
+  identityPoolName: 'myawesomeapp-identitypool',
+});
+```
+
+Identity pools can be configured to allow unauthenticated identities to access the application. This can be specified
+by setting the `unauthenticatedIdentities` boolean property. See [the Cognito
+guide](https://docs.aws.amazon.com/cognito/latest/developerguide/identity-pools.html) to learn more about
+unauthenticated identities.
+
+Identity pools can be configured so that a user authenticating with it will go through a different workflow to bootstrap
+their credentials. There are two different flows for authentication with public providers - basic and enhanced. See [the
+authentication flow page on Cognito](https://docs.aws.amazon.com/cognito/latest/developerguide/authentication-flow.html)
+to learn more about them. Use the `authenticationFlow` property to configure this.
+
+This is the code snippet with these properties configured -
+
+```ts
+new IdentityPool(this, 'myidentitypool', {
+  identityPoolName: 'myawesomeapp-identitypool',
+  allowUnauthenticated: true,
+  authenticationFlow: AuthenticationFlow.ENHANCED,
+});
+```
+
+> These are the defaults that will be part of tsdoc, and not part of the README -
+> * allowUnauthenticated: false
+> * authenticationFlow: `AuthenticationFlow.ENHANCED`
+
+### Identity Providers
+
+Identity pools can be configured with external providers for sign in. These can be from facebook, google, twitter as
+well as, "login with Amazon" and Cognito user pools. Use the `identityProviers` property to configure these providers.
+Learn more about external identity providers at [Cognito's
+documentation](https://docs.aws.amazon.com/cognito/latest/developerguide/external-identity-providers.html). The CDK
+documentation of the `identityProviders` property has more details on how to configure an identity pool with each of
+these external providers.
+
+> Internal Note: IdentityPool-SupportedLoginProviders takes a JSON with all of the providers configured.
```

Comment:
We will provide factory functions for the common use cases, right?

Reply:
The subsequent paragraphs and code snippet is how I propose to model this JSON. This should be sufficient, no?